Public/Src/Cache/ContentStore/Distributed/Redis/RedisContentLocationStore.cs

```diff
@@ -1326,6 +1326,10 @@ public Result<MachineLocation> GetRandomMachineLocation(IReadOnlyList<MachineLoc
             return new Result<MachineLocation>("Could not select a machine location.");
         }
 
+
+        /// <inheritdoc />
+        public bool IsMachineActive(MachineLocation machine) => _idsByLocation.Keys.Contains(machine);

```

Comment:
>_idsByLocation [](start = 64, length = 14)

I think it should be "_idsByLocation.ContainsKey()", because the "Keys" property will acquire a global lock and copy all the keys.